In line number 24 of your output:
The output should be:

^
But your output is: 
terminate called after throwing an instance of 'std::out_of_range'
^

Unmatched character details:
The character in column 1 is supposed to be:
  [Newline (\n) or Carriage Retrun (\r)] ASCII code(13)
but you printed
  [t] ASCII code(116)

